packages/react-native-reanimated/Common/cpp/reanimated/CSS/interpolation/filters/FilterStyleInterpolator.cpp

```diff
@@ -0,0 +1,306 @@
+    return std::nullopt;
+  }
+
+  const auto filtersArray = values.asObject(rt).asArray(rt);
+  const auto filtersCount = filtersArray.size(rt);
+
+  FilterOperations filterOperations;
+  filterOperations.reserve(filtersCount);
+
+  for (size_t i = 0; i < filtersCount; ++i) {
+    const auto filter = filtersArray.getValueAtIndex(rt, i);
+    filterOperations.emplace_back(FilterOperation::fromJSIValue(rt, filter));
+  }
+  return filterOperations;
+}
+
+std::optional<FilterOperations> FilterStyleInterpolator::parseFilterOperations(const folly::dynamic &values) {
+  if (values.empty()) {
+    return std::nullopt;
+  }
+
+  const auto &filtersArray = values;
+  const auto filtersCount = filtersArray.size();
+
+  FilterOperations filterOperations;
+  filterOperations.reserve(filtersCount);
+
+  for (size_t i = 0; i < filtersCount; ++i) {
+    const auto &filter = filtersArray[i];
+    filterOperations.emplace_back(FilterOperation::fromDynamic(filter));
+  }
+  return filterOperations;
+}
+
+std::shared_ptr<FilterKeyframe> FilterStyleInterpolator::createFilterKeyframe(
+    const double fromOffset,
+    const double toOffset,
+    const std::optional<FilterOperations> &fromOperationsOptional,
+    const std::optional<FilterOperations> &toOperationsOptional) const {
+  // If nullopt is passed, return values as is (we will have to read the
+  // filter value from the view style later on and create the new keyframe
+  // then)
+  if (!fromOperationsOptional.has_value() || !toOperationsOptional.has_value()) {
+    return std::make_shared<FilterKeyframe>(
+        FilterKeyframe{fromOffset, toOffset, fromOperationsOptional, toOperationsOptional});
+  }
+
+  const auto [fromOperations, toOperations, isDiscrete] =
+      createFilterInterpolationPair(fromOperationsOptional.value(), toOperationsOptional.value());
+  return std::make_shared<FilterKeyframe>(
+      FilterKeyframe{fromOffset, toOffset, fromOperations, toOperations, isDiscrete});
+}
+
+std::tuple<FilterOperations, FilterOperations, bool> FilterStyleInterpolator::createFilterInterpolationPair(
+    const FilterOperations &fromOperations,
+    const FilterOperations &toOperations) const {
+  FilterOperations fromOperationsResult, toOperationsResult;
+  size_t i = 0;
+  bool isDiscrete = true;
+
+  size_t fromOperationsSize = fromOperations.size();
+  size_t toOperationsSize = toOperations.size();
+  size_t minSize = std::min(fromOperationsSize, toOperationsSize);
+
+  while (i < minSize) {
+    const auto &fromOperation = fromOperations[i];
+    const auto &toOperation = toOperations[i];
+
+    // Types match directly
+    if (fromOperation->type == toOperation->type) {
+      fromOperationsResult.emplace_back(fromOperation);
+      toOperationsResult.emplace_back(toOperation);
+      i++;
+    } else {
+      // If the operation do not match, we need to set keyframe property isDiscrete to false
+      isDiscrete = false;
+      return std::make_tuple(fromOperations, toOperations, isDiscrete);
```

Comment:
```suggestion
      return std::make_tuple(fromOperations, toOperations, true);
```

Can be inlined. Also please assign the opposite value here - `true` instead of `false`.